registry_provider_platform.go

```diff
@@ -0,0 +1,232 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+)
+
+// Compile-time proof of interface implementation
+var _ RegistryProviderPlatforms = (*registryProviderPlatforms)(nil)
+
+// RegistryProviderPlatforms describes the registry provider platform methods supported by the Terraform Enterprise API.
+//
+// TFE API docs: https://www.terraform.io/cloud-docs/api-docs/private-registry/provider-versions-platforms#private-provider-versions-and-platforms-api
+type RegistryProviderPlatforms interface {
+	// Create a provider platform for an organization
+	Create(ctx context.Context, versionID RegistryProviderVersionID, options RegistryProviderPlatformCreateOptions) (*RegistryProviderPlatform, error)
+
+	// List all provider platforms for a single version
+	List(ctx context.Context, versionID RegistryProviderVersionID, options *RegistryProviderPlatformListOptions) (*RegistryProviderPlatformList, error)
+
+	// Read a provider platform by ID
+	Read(ctx context.Context, platformID RegistryProviderPlatformID) (*RegistryProviderPlatform, error)
+
+	// Delete a provider platform
+	Delete(ctx context.Context, platformID RegistryProviderPlatformID) error
+}
+
+// registryProviders implements RegistryProviders
+type registryProviderPlatforms struct {
+	client *Client
+}
+
+// RegistryProviderPlatform represents a registry provider platform
+type RegistryProviderPlatform struct {
+	ID       string `jsonapi:"primary,registry-provider-platforms"`
+	OS       string `jsonapi:"attr,os"`
+	Arch     string `jsonapi:"attr,arch"`
+	Filename string `jsonapi:"attr,filename"`
+	Shasum   string `jsonapi:"attr,shasum"`
+
+	// Relations
+	RegistryProviderVersion *RegistryProviderVersion `jsonapi:"relation,registry-provider-version"`
+
+	// Links
+	Links map[string]interface{} `jsonapi:"links,omitempty"`
+}
+
+// RegistryProviderPlatformID is the multi key ID for identifying a provider platform
+type RegistryProviderPlatformID struct {
+	RegistryProviderVersionID
+	OS   string `jsonapi:"attr,os"`
+	Arch string `jsonapi:"attr,arch"`
```

Comment:
I don't think these fields need encoding metadata because they will not be encoded in a jsonapi request.